### PR TITLE
Handle null inputs to WriteLineAsync correctly.

### LINKED ISSUE #14563: Failures in TextWriter and StreamWriter to handle null inputs to WriteLineAsync
If null is passed to `TextWriter.WriteLineAsync(char[])` it will produce no effect. By analogy to the non-async method it should write a newline as if `TextWriter.WriteLineAsync()` had been called.

If null is passed to `StreamWriter.WriteLineAsync(string)` it will throw a `NullReferenceException` or fail an assertion in Debug. Likewise it should behave as if `StreamWriter.WriteLine()` had been called.

### PR DESCRIPTION
**Pull request by JonHanna**
Fixes #14563

### CODE REVIEW

File: src/System.Runtime.Extensions/tests/System/IO/StreamWriterTests.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class StreamWriterTests
```
Comment (ianhays) [inline src/System.Runtime.Extensions/tests/System/IO/StreamWriterTests.cs]: The existing tests for StreamWriter are in the System.IO assembly.

Comment (JonHanna) [inline src/System.Runtime.Extensions/tests/System/IO/StreamWriterTests.cs]: The test will have to wait then as they won't reference the fixed assembly. Or is there a way around that?

Comment (ianhays) [inline src/System.Runtime.Extensions/tests/System/IO/StreamWriterTests.cs]: >is there a way around that?

Yes, you'll need to add an explicit p2p reference like the ones removed in [this commit](https://github.com/dotnet/corefx/commit/963bbf421720f2d1a321f17360c2ad3c04d3d4da)

Comment (JonHanna) [inline src/System.Runtime.Extensions/tests/System/IO/StreamWriterTests.cs]: Ah of course. That seems obvious now that you said it. *facepalm*.

Comment (ianhays) [inline src/System.Runtime.Extensions/tests/System/IO/StreamWriterTests.cs]: :)

**Comment by ianhays** [comment]
Changes LGTM pending the outcome of the discussion in #14563

**Comment by karelz** [comment]
The discussion #14563 seems to be stuck for 18 days. Can we push it forward, or do we need to wait longer for a consensus?

**Comment by ianhays** [comment]
I'm in favor of taking this change. Since no one seems to be strongly opposed to it, I'm going to merge this.

There are a few reasons why (which are specified in the linked issue), but my main justification for this is that the existing behavior is contrary to the behavior specified in the documentation and I don't want to update the docs to assert this wonky functionality.  The likelihood that we break someone is pretty small here.

**Comment by karelz** [comment]
Thanks @ianhays!

**Pull request merged**